### PR TITLE
Optimize

### PR DESCRIPTION
**Pull request by ghostant-1017**

### CODE REVIEW

**Review by raychu86 — APPROVED** [review]
lgtm

File: ledger/block/src/transactions/confirmed/mod.rs [hunk 1]
```diff
@@ -142,6 +142,15 @@ impl<N: Network> ConfirmedTransaction<N> {
...
     }
 
+    pub fn variant(&self) -> String {
+        match self {
+            Self::AcceptedDeploy(..) => "accepted deploy".to_string(),
+            Self::AcceptedExecute(..) => "accepted execute".to_string(),
+            Self::RejectedDeploy(..) => "rejected deploy".to_string(),
+            Self::RejectedExecute(..) => "rejected execute".to_string(),
+        }
+    }
```
Comment (howardwu) [inline ledger/block/src/transactions/confirmed/mod.rs]: We can save 1 allocation by changing the return type to `&str` and avoid calling `.to_string()`.

Also, I realized there are many functions here (including this one) that can be changed from `pub fn` to `pub const fn`.

**Review by howardwu — APPROVED** [review]
LGTM

File: ledger/block/src/transactions/confirmed/mod.rs [hunk 2]
```diff
@@ -142,8 +142,17 @@ impl<N: Network> ConfirmedTransaction<N> {
         }
     }
 
+    pub const fn variant(&self) -> &str {
```
Comment (howardwu) [inline ledger/block/src/transactions/confirmed/mod.rs:146]:
```suggestion
    /// Returns the human-readable variant of the confirmed transaction. 
    pub const fn variant(&self) -> &str {
```

**Pull request merged**